### PR TITLE
configure: add -02 to flags

### PR DESCRIPTION
**Pull request by williamcroberts**
To fix FORTIFY_SOURCE requiring optimization errors:
/usr/include/features.h:382:4: error: #warning _FORTIFY_SOURCE requires compiling with optimization (-O) [-Werror=cpp]
  382 | #  warning _FORTIFY_SOURCE requires compiling with optimization (-O)
      |    ^~~~~~~
cc1: all warnings being treated as errors
make[1]: *** [Makefile:2451: src/tabrmd.o] Error 1

Add -O2 to the flags.

Fixes: #652

Signed-off-by: William Roberts <william.c.roberts@intel.com>

### CODE REVIEW

**Comment by williamcroberts** [comment]
I see your code cov says it's not working. I wonder if we need CONFIG_EXTRA to include `--enable-defaultflags=no`

**Comment by williamcroberts** [comment]
You're actually picking up a little distro coverage in tpm2-tools CI as well:
https://travis-ci.org/tpm2-software/tpm2-tools/jobs/578608737

File: .travis.yml [hunk 1]
```diff
@@ -42,7 +42,7 @@ env:
 matrix:
   include:
     - compiler: clang
-      env: SCANBUILD="scan-build --status-bugs" CFLAGS=-D_REENTRANT CONFIG_EXTRA=--enable-code-coverage MAKE_TARGET=check
+      env: SCANBUILD="scan-build --status-bugs" CFLAGS=-D_REENTRANT CONFIG_EXTRA="--enable-code-coverage --enable-defaultflags" MAKE_TARGET=check
```
Comment (flihp) [inline .travis.yml]: This adds `--enable-defaultflags` to `CONFIG_EXTRA`. Based on the commit message I was expecting `--disable-defaultflags`. Am I missing something?

Comment (williamcroberts) [reply .travis.yml]: Yeah, their's also an issue with the check for code cov here:
https://travis-ci.org/tpm2-software/tpm2-abrmd/jobs/578610194#L1771

I'll have to update that as well to be a sub-string check.

Comment (williamcroberts) [reply .travis.yml]: face palm, that moment you realize you had it right locally and never updated the branch the PR is based off of.

**Comment by williamcroberts** [comment]
Screenshot from 2019-09-03 10-24-22

Looks like codecov is happy again...

**Comment by flihp** [comment]
thx for the fixups, LGTM

**Pull request merged**